WalletWasabi/Logging/Logger.cs

```diff
@@ -227,6 +227,27 @@ private static void Log(LogLevel level, string message, int additionalEntrySepar
 			}
 		}
 
+		// This method removes the path and file extension.
+		//
+		// Given Wasabi releases are currently built using Windows, the generated assymblies contain
+		// the hardcoded "C:\Users\User\Desktop\WalletWasabi\.......\FileName.cs" string because that
+		// is the real path of the file, it doesn't matter what OS was targeted.
+		// In Windows and Linux that string is a valid path and that means Path.GetFileNameWithoutExtension
+		// can extract the file name but in the case of OSX the same string is not a valid path so, it assumes
+		// the whole string is the file name. 
+		private static string ExtractFileName(string callerFilePath)
+		{
+			var lastSeparatorIndex = callerFilePath.LastIndexOf("\\");
```

Comment:
Null check this pls - return "nofile" if null.

Reply:
It is a `private` method and as such it should never be called with invalid parameters. Btw, it is never called with `null` because the caller checks for null before calling it, as it should always be.

Reply:
I can pass null as a parameter to 

`public static void LogInfo(string message, [CallerFilePath] string callerFilePath = ""...`

Reply:
I can pass null to callerFilePath argument and it is public:
`public static void LogInfo(string message, [CallerFilePath] string callerFilePath = ""`

Reply:
No, it cannot. This method is called by the following line:

```c#
var category = string.IsNullOrWhiteSpace(callerFilePath) ? "" : $"{ExtractFileName(callerFilePath)} ({callerLineNumber})";
```

Reply:
Right, how could I missed that...